src/mgmt/srv_drpc.c

```diff
@@ -2033,6 +2038,7 @@ ds_mgmt_drpc_smd_list_devs(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 
 	for (i = 0; i < resp->n_devices; i++) {
 		if (resp->devices[i] != NULL) {
+			ds_mgmt_smd_free_dev_ctrlr(resp->devices[i]);
 			ds_mgmt_smd_free_dev(resp->devices[i]);
```

Comment:
What about freeing `dev->ctrlr` itself? I'm not sure I understand the motivation for breaking this cleanup into two functions.

Reply:
yes I'm missed that thank you
separating them out to facilitate handlers that populate ctrlrs from those that don't
it resolves the segfault that was being seen in ftest